docs/src/examples/basics.md

```diff
@@ -0,0 +1,206 @@
+```
+![roadway basic manual](roadway_basic_manual.svg)
+
+The result remains the same. The velocity arrows point to the location at which
+the vehicle would be 1 second in the future.
+
+Setting `AutoViz.rendermode` to `:fancy`, the rectangles are replaced by SVG
+images of cars (or pedestrians). 
+
+```@example intro
+# AutoViz.rendermode = :fancy # cannot assign variables in other modules
+@eval(AutoViz, rendermode = :fancy)
+snapshot = render([roadway, scene], canvas_height=120)
+write("roadway_fancy.svg", snapshot) # hide
+```
+![roadway fancy](roadway_fancy.svg)
+
+Which is shorthand for
+
+```@example intro
+renderables = [
+    roadway, (FancyCar(car=scene[i]) for i in 1:3)..., FancyPedestrian(ped=scene[4])
+]
+snapshot = render(renderables, canvas_height=120)
+write("roadway_fancy_manual.svg", snapshot) # hide
+```
+![roadway fancy manual](roadway_fancy_manual.svg)
+
+A third visualization mode is available in the form of arrow cars, in which the
+arrow indicates the heading direction of the car but does not scale with speed.
+
+```@example intro
+renderables = [
+    roadway, (ArrowCar(car=scene[i]) for i in 1:3)..., FancyPedestrian(ped=scene[4])
+]
+snapshot = render(renderables, canvas_height=120)
+write("roadway_arrow.svg", snapshot) # hide
+```
+![roadway arrow](roadway_arrow.svg)
+
+
+## Vehicle Colors
+
+By default, the `render` function generates a random color for each entity based
+on its ID using the `id_to_color()` function.
+However, vehicle colors can also be assigned explicitly:
+
+```@example intro
+colors = [colorant"red", colorant"blue", colorant"green"]
+renderables = [
+    roadway,
+    (FancyCar(car=scene[i], color=colors[i]) for i in 1:3)...,
+    FancyPedestrian(ped=scene[4], color=colorant"yellow")
+]
+snapshot = render(renderables, canvas_height=120)
+write("scene_custom_colors.svg", snapshot) # hide
+```
+![cars with custom colors](scene_custom_colors.svg)
+
+
+
+## Simulation and Animations
+
+We can simulate the scenario over time and visualize the results using `Reel`
+(based on `ffmpeg`).
+
+```@example intro
+using Reel
+
+timestep = 0.1
+nticks = 50
+
+models = Dict((i => Tim2DDriver(timestep) for i in 1:3))  # car models
+# TODO: use a different model for pedestrian
+models[42] = Tim2DDriver(timestep)  # pedestrian model
+
+scenes = simulate(scene, roadway, models, nticks, timestep)
+
+animation = roll(fps=1.0/timestep, duration=nticks*timestep) do t, dt
+    i = Int(floor(t/dt)) + 1
+    render([roadway, scenes[i]])
+end
+
+write("roadway_animated.gif", animation)
+```
+![animated roadway](roadway_animated.gif)
+
+Alternatively, the scene can also be visualized interactively using `Interact`
+
+```julia
+using Interact
+using Reel
+using Blink
+
+w = Window()
+viz = @manipulate for step in 1 : length(scenes)
+    render([roadway, scenes[step]])
+end
+body!(w, viz)
+```
+
+You can use Reactive to have them endlessly drive in real time in your browser.
+
+### TODO: is this a thing? If yes, provide snippet
```

Comment:
I've never used Reactive for visualizing simulations, is this a thing?

Reply:
I never used it either

Reply:
ok, then maybe I'll leave it at a honorable mention of the package instead of providing a working code example.